internal/controller/entra/main.go

```diff
@@ -134,41 +143,59 @@ func (c *Controller) IsStaffMember(ctx context.Context, username types.Username)
 
 func (c *Controller) SendInvite(ctx context.Context, email string, sponsor types.Sponsor) error {
 
-	// check if user exists in entra, if yes, don't send invite
+	// check if user exists in entra, if not, send an invite; if yes, custom notification
 	user, err := c.userData(ctx, types.Username(email))
-	if err != nil {
+	if err != nil && strings.Contains(err.Error(), "does not exist") {
+		// we only want to return the error if it's not because the user doesn't exist, otherwise we can't invite them
+
+		requestBody := graphmodels.NewInvitation()
+		invitedUserEmailAddress := email
+		inviteRedirectUrl := config.EntraInviteRedirectURL()
+		sendInvitationMessage := true
+
+		requestBody.SetInvitedUserEmailAddress(&invitedUserEmailAddress)
+		requestBody.SetInviteRedirectUrl(&inviteRedirectUrl)
+		requestBody.SetSendInvitationMessage(&sendInvitationMessage)
+
+		message := ""
+		if sponsor.ChosenName != "" {
+			message = "You have been invited to join the UCL ARC Services Portal by " + string(sponsor.ChosenName)
+		} else {
+			message = "You have been invited to join the UCL ARC Services Portal by " + string(sponsor.Username)
+		}
+		messageInfo := graphmodels.NewInvitedUserMessageInfo()
+
+		messageInfo.SetCustomizedMessageBody(&message)
+		requestBody.SetInvitedUserMessageInfo(messageInfo)
+
+		_, err = c.client.Invitations().Post(ctx, requestBody, nil)
+		if err != nil {
+			if strings.Contains(err.Error(), "already exists") {
```

Comment:
how do we get here if we've checked if the user exists at the top of this function?

Reply:
tried with ccaetyo@... & tom.a.young@...
both triggered that path, but I don't see them in our TRE tenant? see img 
So I don't entirely understand why but it happens
<img width="437" height="225" alt="image" src="https://github.com/user-attachments/assets/0ef61cf7-669c-41b4-9714-b0b568ab0a93" />

It may be a peculiarity because in development we're using two different tenants?